openwisp_utils/utils.py

```diff
@@ -62,3 +65,17 @@ def print_color(string, color_name, end='\n'):
     }
     color = color_dict.get(color_name, '0')
     print(f'\033[{color}m{string}\033[0m', end=end)
+
+
+def retryable_request(method, timeout=(4, 8), max_retries=3, **kwargs):
+    request_session = requests.Session()
+    retries = Retry(
+        total=max_retries,
+        backoff_factor=1,
+        status_forcelist=[429, 500, 502, 503, 504],
```

Comment:
This new utility function will be useful, please allow to custoimize the arguments passed to `Retry()`, eg:

- backoff_factor
- backoff_jitter
- status_forcelist
- allowed_methods
- add another argument called retry_kwargs=None

Then I recommend doing something  like:

```python
if retry_kwargs is None:
    retry_kwargs = {}
retry_kwargs.update(dict(
    total=max_retries,
    backoff_factor=backoff_factor,
    backoff_jitter=backoff_jitter,
    status_forcelist=status_forcelist,
    allowed_methods=allowed_methods
))
retries = Retry(**retry_kwargs)
```

This will allow us to reuse this function in other modules because as you know it's a common need.
Please add this function to the README and make sure there's at least one unit test dedicated to this function only.